compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CommonContextMenuArea.kt

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.text
+
+import androidx.compose.foundation.ComposeFoundationFlags
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.contextmenu.ContextMenuArea
+import androidx.compose.foundation.contextmenu.ContextMenuScope
+import androidx.compose.foundation.contextmenu.ContextMenuState
+import androidx.compose.foundation.contextmenu.close
+import androidx.compose.foundation.text.contextmenu.data.TextContextMenuKeys
+import androidx.compose.foundation.text.contextmenu.internal.ProvideDefaultPlatformTextContextMenuProviders
+import androidx.compose.foundation.text.contextmenu.modifier.textContextMenuGestures
+import androidx.compose.foundation.text.input.internal.selection.TextFieldSelectionState
+import androidx.compose.foundation.text.input.internal.selection.contextMenuBuilder
+import androidx.compose.foundation.text.selection.SelectionManager
+import androidx.compose.foundation.text.selection.TextFieldSelectionManager
+import androidx.compose.foundation.text.selection.contextMenuBuilder
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import kotlin.jvm.JvmInline
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal fun CommonContextMenuArea(
+    manager: TextFieldSelectionManager,
+    content: @Composable () -> Unit
+) {
+    if (ComposeFoundationFlags.isNewContextMenuEnabled) {
+        ProvideDefaultPlatformTextContextMenuProviders(manager.contextMenuAreaModifier, content)
+    } else {
+        val state = remember { ContextMenuState() }
+        val coroutineScope = rememberCoroutineScope()
+        val menuItemsAvailability = remember { mutableStateOf(MenuItemsAvailability.None) }
```

Comment:
This `remember` doesn't have any keys. 
The avalability state is synced in every `onOpenGesture` and there could be a delay in web due to async Clipboard check (a Promise).
So until the Promise is resolved there could a moment when the `menuItemsAvailability` contains an outdated state - the state from the prvious `onOpenGesture` sync.

I think we need a key for `remember`, so the state won't contain any outdated values.
I think that the state can be reset to `MenuItemsAvailability.None` when the menu becomes `Status.Closed`.

WDYT?